pkg/cmd/pulumi/state_move.go

```diff
@@ -0,0 +1,213 @@
+// 			ctx := cmd.Context()
+
+// 			if sourceStackName == "" && destStackName == "" {
+// 				return errors.New("at least one of --source or --dest must be provided")
+// 			}
+// 			// TODO: make sure to load the source stack even if it is from a different project
+// 			sourceStack, err := requireStack(ctx, sourceStackName, stackLoadOnly, display.Options{
+// 				Color:         cmdutil.GetGlobalColorization(),
+// 				IsInteractive: true,
+// 			})
+// 			if err != nil {
+// 				return err
+// 			}
+// 			// TODO: make sure to load the dest stack even if it is from a different project.
+// 			destStack, err := requireStack(ctx, destStackName, stackLoadOnly, display.Options{
+// 				Color:         cmdutil.GetGlobalColorization(),
+// 				IsInteractive: true,
+// 			})
+// 			if err != nil {
+// 				return err
+// 			}
+
+// 			return stateMove.Run(ctx, sourceStack, destStack, args)
+// 		}),
+// 	}
+
+// 	cmd.Flags().StringVarP(&sourceStackName, "source", "", "", "The name of the stack to move resources from")
+// 	cmd.Flags().StringVarP(&destStackName, "dest", "", "", "The name of the stack to move resources to")
+
+// 	return cmd
+// }
+
+func resourceMatches(res *resource.State, args []string) bool {
+	for _, arg := range args {
+		if string(res.URN) == arg {
+			return true
+		}
+	}
+	return false
+}
+
+func breakDependencies(res *resource.State, resourcesToMove map[string]*resource.State) {
+	j := 0
+	for _, dep := range res.Dependencies {
+		if _, ok := resourcesToMove[string(dep)]; !ok {
+			res.Dependencies[j] = dep
+			j++
+		}
+	}
+	res.Dependencies = res.Dependencies[:j]
+	for k, propDeps := range res.PropertyDependencies {
+		j = 0
+		for _, propDep := range propDeps {
+			if _, ok := resourcesToMove[string(propDep)]; !ok {
+				propDeps[j] = propDep
+				j++
+			}
+		}
+		res.PropertyDependencies[k] = propDeps[:j]
+	}
+	if _, ok := resourcesToMove[string(res.DeletedWith)]; !ok {
+		res.DeletedWith = ""
+	}
+}
+
+func rewriteURNs(res *resource.State, dest backend.Stack) {
+	// TODO: rewrite project name
+	res.URN = res.URN.RenameStack(dest.Ref().Name())
+	if res.Provider != "" {
+		res.Provider = string(urn.URN(res.Provider).RenameStack(dest.Ref().Name()))
+	}
+	for k, dep := range res.Dependencies {
+		res.Dependencies[k] = dep.RenameStack(dest.Ref().Name())
+	}
+	for k, propDeps := range res.PropertyDependencies {
+		for j, propDep := range propDeps {
+			res.PropertyDependencies[k][j] = propDep.RenameStack(dest.Ref().Name())
+		}
+	}
+	if res.DeletedWith != "" {
+		res.DeletedWith = res.DeletedWith.RenameStack(dest.Ref().Name())
+	}
+}
+
+func (cmd *stateMoveCmd) Run(ctx context.Context, source backend.Stack, dest backend.Stack, args []string) error {
+	sourceSnapshot, err := source.Snapshot(ctx, stack.DefaultSecretsProvider)
+	if err != nil {
+		return err
+	}
+	destSnapshot, err := dest.Snapshot(ctx, stack.DefaultSecretsProvider)
+	if err != nil {
+		return err
+	}
```

Comment:
Wonder if it's worth verifying integrity before we start?

Reply:
Probably a good idea.  We should always be in a good state, but that's not guaranteed :heavy_check_mark: